library/std/canon.qs

```diff
@@ -438,5 +438,45 @@ namespace Microsoft.Quantum.Canon {
         }
     }
 
+    /// # Summary
+    /// Applies a unitary operation on the target register,
+    /// controlled on a state specified by a given bit mask.
+    ///
+    /// # Input
+    /// ## bits
+    /// The bit string to control the given unitary operation on.
+    /// ## oracle
+    /// The unitary operation to be applied on the target register.
+    /// ## targetRegister
+    /// The target register to be passed to `oracle` as an input.
+    /// ## controlRegister
+    /// A quantum register that controls application of `oracle`.
+    ///
+    /// # Remarks
+    /// The pattern given by `bits` may be shorter than `controlRegister`,
+    /// in which case additional control qubits are ignored (that is, neither
+    /// controlled on $\ket{0}$ nor $\ket{1}$).
+    /// If `bits` is longer than `controlRegister`, an error is raised.
+    ///
+    /// For example, `bits = [0,1,0,0,1]` means that `oracle` is applied if and only if `controlRegister`
+    /// is in the state $\ket{0}\ket{1}\ket{0}\ket{0}\ket{1}$.
+    operation ApplyControlledOnBitString<'T>(
+        bits: Bool[],
+        oracle: ('T => Unit is Adj + Ctl),
+        controlRegister: Qubit[],
+        targetRegister: 'T): Unit is Adj + Ctl {
```

Comment:
nit: I would rename `targetRegister` to `target` since it is not really a register, it is a generic that represents the input for the oracle.

Reply:
It could be a register :) and the comments refer to it as the register, but I see your point. I may need to update 'controlled on int' as well.

Reply:
Updated.